mysql-test/suite/wsrep/t/wsrep_set_var_extension.test

```diff
@@ -0,0 +1,130 @@
+# === Purpose ===
+#
+# This test verifies that the following wsrep session variables are hint updatable.
+#
+# 1. wsrep_sync_wait
+# 2. wsrep_dirty_reads
+# 3. wsrep_trx_fragment_size
+# 4. wsrep_trx_fragment_unit
+#
+# === References ===
+#
+# PXC-3743: Make wsrep session variables updatable through SET_VAR() hint
+
+--source include/have_wsrep_provider.inc
+
+# Auxiliary SP which returns variable value before hint, with hint and after hint applying
+DELIMITER \\; 
+
+CREATE PROCEDURE test_hint (hint_str VARCHAR(255), var_str VARCHAR(64))
+BEGIN
+
+SET @orig_q= CONCAT("SELECT VARIABLE_VALUE INTO @before_val FROM performance_schema.session_variables where VARIABLE_NAME = '",  var_str, "'");
+
+SET @hint_q= CONCAT("SELECT /*+ ", hint_str,
+                    "*/ VARIABLE_VALUE" ,
+                    " INTO @hint_val FROM performance_schema.session_variables where VARIABLE_NAME = '",  var_str, "'");
+
+SET @after_q= CONCAT("SELECT VARIABLE_VALUE INTO @after_val FROM performance_schema.session_variables where VARIABLE_NAME = '",  var_str, "'");
+
+PREPARE orig_stmt FROM @orig_q;
+PREPARE hint_stmt FROM @hint_q;
+PREPARE after_stmt FROM @after_q;
+
+EXECUTE orig_stmt;
+EXECUTE hint_stmt;
+EXECUTE after_stmt;
+
+SELECT  hint_str;
+SELECT @before_val, @hint_val, @after_val;
+
+DEALLOCATE PREPARE orig_stmt;
+DEALLOCATE PREPARE hint_stmt;
+DEALLOCATE PREPARE after_stmt;
+
+END\\
+
+DELIMITER ;\\
+
+--echo #########################################
+--echo # Test SET_VAR hint for wsrep variables #
+--echo #########################################
+
+--echo
+--echo # Test 1: wsrep_sync_wait
+CALL test_hint("SET_VAR(wsrep_sync_wait=4)", "wsrep_sync_wait");
+
+SET wsrep_sync_wait=2;
+CALL test_hint("SET_VAR(wsrep_sync_wait=4)", "wsrep_sync_wait");
+
+--assert (`SELECT @before_val != @hint_val`)
+--assert (`SELECT @hint_val != @after_val`)
+--assert (`SELECT @before_val = @after_val`)
+
+--assert (`SELECT /*+ SET_VAR(wsrep_sync_wait=10)*/ @@wsrep_sync_wait = 10`)
+SET wsrep_sync_wait=default;
+
+--echo
+--echo # Test 2: wsrep_dirty_reads
+CALL test_hint("SET_VAR(wsrep_dirty_reads=1)", "wsrep_dirty_reads");
+
+SET wsrep_dirty_reads=1;
+CALL test_hint("SET_VAR(wsrep_dirty_reads=0)", "wsrep_dirty_reads");
+
+--assert (`SELECT @before_val != @hint_val`)
+--assert (`SELECT @hint_val != @after_val`)
+--assert (`SELECT @before_val = @after_val`)
+
+--assert (`SELECT /*+ SET_VAR(wsrep_dirty_reads=1)*/ @@wsrep_dirty_reads = 1`)
+SET wsrep_dirty_reads=default;
+
+
+--echo
+--echo # Test 2: wsrep_dirty_reads
```

Comment:
Does this check/test differ from the one above?

Reply:
Thanks for catching this. This is a duplicate. Removed now.